File: backend/src/main/java/com/petrovic/fooddeliveryapi/models/Product.java
package com.petrovic.fooddeliveryapi.models;

import lombok.*;

import javax.persistence.*;
import java.util.UUID;

@Entity(name = "products")
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Product {
    @Id
    private String id;
    private String name;
    private String description;
    private String imageUrl;
    private float price;
    private boolean isAvailable;

    @PrePersist
    public void prePersist() {
        if (id == null) {
            id = UUID.randomUUID().toString();
        }
    }
}

File: frontend/src/models/product.ts
export class Product {
    id: number;
    available: boolean;
    description: string;
    imageUrl: string;
    name: string;
    price: number;

    constructor(
        id: number,
        available: boolean,
        description: string,
        imageUrl: string,
        name: string,
        price: number
    ) {
        this.id = id;
        this.available = available;
        this.description = description;
        this.imageUrl = imageUrl;
        this.name = name;
        this.price = price;
    }
}
